craft_parts/executor/executor.py

```diff
@@ -165,6 +165,7 @@ def clean(self, initial_step: Step, *, part_names: list[str] | None = None) -> N
                 for stage_dir in self._project_info.stage_dirs.values():
                     if stage_dir.exists():
                         shutil.rmtree(stage_dir)
+                shutil.rmtree(self._project_info.backstage_dir)
```

Comment:
let's leave cleaning for another body of work. This feels like the wrong place to do this

Reply:
That was leftover from the experimental branch work. Removed.